pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -96,9 +98,16 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 		}
 	}
 
-	err = validateIdentityUrl(doc.OpenShiftCluster, identityURL, isCreate)
-	if err != nil {
-		return nil, err
+	// Don't persist identity parameters in non-wimi clusters
+	if doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil || doc.OpenShiftCluster.Identity != nil {
```

Comment:
To be consistent, the current way of determining the WIMI cluster is:-
`doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil && doc.OpenShiftCluster.PlatformWorkloadIdentityProfile != nil` all other are considered to non-miwi.

Reply:
Not checking `doc.OpenshiftCluster.Identity` to determine MIWI cluster seems pretty problematic - for example, let's say a customer specifies the workload identities but forgets to specify the cluster MSI. The above check would pass even though the cluster doesn't have a cluster MSI.

I'll write a separate function to determine if an `OpenShiftCluster` is MIWI or not so we have a single source of truth.

Reply:
@niontive @rajdeepc2792 we already have a shared function in progress, and the logic is currently `doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil || doc.OpenShiftCluster.Identity != nil`. The idea with || is that we want to account for cases where the cluster doc is not exactly as expected.

Reply:
@cadenmarchese I'll revert my change for a shared function and use `doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil || doc.OpenShiftCluster.Identity != nil`. I'll add a TODO to use as shared function once it's implemented so we don't forget.